src/main/java/org/broadinstitute/consent/http/service/dao/DacServiceDAO.java

```diff
@@ -0,0 +1,74 @@
+package org.broadinstitute.consent.http.service.dao;
+
+import com.google.inject.Inject;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Objects;
+import org.broadinstitute.consent.http.db.DaaDAO;
+import org.broadinstitute.consent.http.models.Dac;
+import org.broadinstitute.consent.http.models.DataAccessAgreement;
+import org.broadinstitute.consent.http.util.ConsentLogger;
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.core.statement.Update;
+
+public class DacServiceDAO implements ConsentLogger {
+
+  private final Jdbi jdbi;
+  private final DaaDAO daaDAO;
+
+  @Inject
+  public DacServiceDAO(Jdbi jdbi, DaaDAO daaDAO) {
+    this.jdbi = jdbi;
+    this.daaDAO = daaDAO;
+  }
+
+  public void deleteDacAndDaas(Dac dac)
+      throws IllegalArgumentException, SQLException {
+    // fail fast
+    if (Objects.isNull(dac)) {
+      throw new IllegalArgumentException("Invalid DAC");
+    }
+    List<DataAccessAgreement> daas = daaDAO.findAll();
+    List<DataAccessAgreement> filteredDaas = daas.stream()
+        .filter(daa -> daa.getInitialDacId().equals(dac.getDacId())).toList();
```

Comment:
See comment where `filteredDaas` is used - we don't need to filter them nor do we need to get all DAAs to populate this list.